src/nemos/utils.py

```diff
@@ -400,9 +401,7 @@ def row_wise_kron(a, c):
     return K
 
 
-def check_invalid_entry(
-    pytree: Union[FeaturePytree, jnp.ndarray], pytree_name: str
-) -> None:
+def check_invalid_entry(pytree: Any, pytree_name: str) -> Tuple[Any, Union[None, str]]:
```

Comment:
like I mentioned in other comment, I don't think it's great to have this just return the error message, without doing anything with it -- too easy to forget  to raise it.

Alternative proposal: we have a `_check_invalid_entry` that is basically this and then a `warn_invalid_entry` and `error_invalid_entry`, which do the two versions of this: checks/warns/drops/returns valid entries, and checks/raise exception.

Thoughts?

Reply:
Or, if `warn_invalid_entry` shouldn't do dropping, then a separate function for the dropping? probably should be a helper function, since it will be useful.

Reply:
we need to merge the invalid entries of X and y, should we pass both to the checker? I think we should and I think it should implement something like _check_invalid_entry(*args), _warn_invalid_entry(*args), even though I would like the error message to be variable specific

Reply:
New design:
1) func check raises error
2) func drop invalid (warn and drop) receives all *args, pass (arg, argname).
3) fit needs drop and predict none of the two. Test that predict with nan has expected behavior